### ISSUE TITLE
Advanced Search Filters

### ISSUE THREAD

**Issue by gayatri-p786** [akshitagupta15june/PetMe-Prod#21]
Enhance the "Adopting a Pet" page by adding an advanced search feature. This will help users find their ideal pets more easily by filtering based on various criteria.
**Filters:**
Type of Animal (Cat, Dog, Bird, Rabbit, Reptile, Other)
Breed (Specific breeds, Mixed breed)
Age (Puppy/Kitten, Young, Adult, Senior)
Size (Small, Medium, Large)
Gender (Male, Female)
Location (Distance, Specific city/region)
Adoption Fee (Free, Under $50, $50-$100, Over $100)

**Comment by gayatri-p786** [comment]
Hi, Can you please assign me this issue? I want to contribute to enhanced user experience.

**Comment by cb7chaitanya** [comment]
You would have to setup an API endpoint for that in the backend and frontend both then, willing to work on this?

**Comment by vineet1202** [comment]
@cb7chaitanya Hi, I'd like to work on this issue too. If @gayatri-p786 is open to collaboration, could you please assign it to both of us?

**Comment by gayatri-p786** [comment]
Yes, I am willing to work and also collaborate on this

**Comment by cb7chaitanya** [comment]
Assigning it to both of you, work together and split your tasks, also feel free to ping me in discord if you need any guidance @gayatri-p786 @vineet1202

**Comment by gayatri-p786** [comment]
Thank you @cb7chaitanya! @vineet1202 would you like to connect on discord first to discuss and then move forward from there? My discord username: gayatri3538

**Comment by cb7chaitanya** [comment]
Yes @gayatri-p786 we are setting up a meeting with all new assignees, you can join then would explain the flow of the application

**Comment by cb7chaitanya** [comment]
Start working on this guys, request any new API endpoints you want, if you want them @gayatri-p786 @vineet1202
